shim/src/realm.js

```diff
@@ -1,61 +1,226 @@
+        [ImportHook]: undefined,
+        [ImportMetaHook]: undefined,
+        // ---> divering to create the internal shim iframe
+        [ShimSandbox]: createSandbox(),
+    };
+    if (intrinsics === undefined) {
+        CreateIntrinsics(realmRec);
+    } else {
+        // 1. Assert: In this case, _intrinsics_ must be a Record with field names listed in column one of Table 7.
+        realmRec[Intrinsics] = intrinsics;
+    }
+    return realmRec;
+}
+
+function InvokeDirectEvalHook(realmRec, x) {
+    // 1. Assert: realm is a Realm Record.
+    const fn = realmRec[EvalHook];
+    if (fn === undefined) return x;
+    assert(IsCallable(fn) === true);
+    return fn.call(undefined, x);
+}
+
+// <!-- es6num="18.2.1.1" -->
+function PerformEval(x, evalRealm, strictCaller, direct) {
+    assert(direct === false ? strictCaller === false : true);
+    // realm spec segment begins
+    if (direct === true) {
+        x = InvokeDirectEvalHook(x, evalRealm);
+    }
+    // realm spec segment ends
+    if (typeof x !== 'string') return x;
+    // ---> divering
+    const sandbox = getSandbox(evalRealm);
+    return evaluate(x, sandbox);
+}
+
 export default class Realm {
 
-    constructor(options = {}) {
-        const { thisValue, globalObj } = options;
-        const sandbox = createSandbox(globalObj, thisValue);
-        sanitize(sandbox);
-        Object.assign(sandbox, getEvaluators(sandbox));
-        // TODO: assert that RealmToSandbox does not have `this` entry
-        RealmToSandbox.set(this, sandbox);
-        sandbox.globalProxy = new Proxy(sandbox, proxyHandler);
-        this.init();
+    constructor(options) {
+        const O = this;
+        const parentRealm = getCurrentRealmRecord();
+        const opts = Object(options);
+        let importHook = opts.importHook;
+        if (importHook === "inherit") {
+            importHook = parentRealm[ImportHook];
+        } else if (importHook !== undefined && IsCallable(importHook) === false) throw new TypeError();
+        let importMetaHook  = opts.importMetaHook;
+        if (importMetaHook === "inherit") {
+            importMetaHook = parentRealm[ImportMetaHook];
+        } else if (importMetaHook !== undefined && IsCallable(importMetaHook) === false) throw new TypeError();
+        let evalHook = opts.evalHook;
+        if (evalHook === "inherit") {
+            evalHook = parentRealm[EvalHook];
+        } else if (evalHook !== undefined && IsCallable(evalHook) === false) throw new TypeError();
+        let isDirectEvalHook = opts.isDirectEvalHook;
+        if (isDirectEvalHook === "inherit") {
+            isDirectEvalHook = parentRealm[IsDirectEvalHook];
+        } else if (isDirectEvalHook !== undefined && IsCallable(isDirectEvalHook) === false) throw new TypeError();
+        let intrinsics = opts.intrinsics;
+        if (intrinsics === "inherit") {
+            intrinsics = parentRealm[Intrinsics];
+        } else if (intrinsics !== undefined) throw new TypeError();
```

Comment:
At the moment it can only be `undefined` or the string value `"inherit"`, anything else will throw. Eventually we want to allow them to provide a reference to another intrinsics object, in which case we will have to make some validations... but we haven't get to that just yet.